oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java

```diff
@@ -38,27 +41,50 @@
 @Slf4j
 public class StorageEsInstaller extends ModelInstaller {
     private final Gson gson = new Gson();
-
     private final StorageModuleElasticsearchConfig config;
     protected final ColumnTypeEsMapping columnTypeEsMapping;
 
+    /**
+     * The relation is between physical table and the specific structure definition.
+     */
+    private final ConcurrentHashMap<String, Map<String, Object>> tables;
```

Comment:
I think should use `StoragePartitioner` to hold, and put a class called `PhysicalIndex` the value of `StoragePartitioner#tableRelations`. 
According to your code style, you should isolate these two.

Reply:
Just merge together and can be isolatedd again

Reply:
This is not my point. I mean you keep related data in different places.

Reply:
StoragePartitioner is managing the metadata of ES storage module, then all related cache and metadata should be there.